docs/how-to/write-devices.md

```diff
@@ -0,0 +1,70 @@
+# Write Devices for Blueapi
+
+:::{seealso}
+[Home of Plans and Devices](../explanations/extension-code.md) for information about where device code usually lives.
+:::
+
+
+## Format
+
+:::{seealso}
+[Dodal](https://github.com/DiamondLightSource/dodal) for many more examples
+:::
+
+Devices are made using the [dodal](https://github.com/DiamondLightSource/dodal) style available through factory functions like this:
+
+```python 
+from my_facility_devices import MyTypeOfDetector
+
+def my_detector(name: str) -> MyTypeOfDetector:
+    return MyTypeOfDetector(name, {"other_config": "foo"})
+```
+
+The return type annotation `-> MyTypeOfDetector` is required as blueapi uses it to determine that this function creates a device. Meaning you can have a Python file where only some functions create devices and they will be automatically picked up. Similarly, these functions can be organized per-preference into files. 
+
+The device is created via a function rather than a global to preserve side-effect-free imports. Each device must have its own factory function.
```

Comment:
Should: defer to dodal's docs on writing device classes and beamlines, and ensure they are up to date?

Reply:
Agreed, see other discussions, I think we should migrate some of this to dodal as a separate PR

Reply:
dodal already has docs on creating a beamline and making a device class. 

https://diamondlightsource.github.io/dodal/main/how-to/create-beamline.html
and
https://diamondlightsource.github.io/dodal/main/how-to/make-new-ophyd-async-device.html

We should defer to these and fix those pages. The former should be updated to use device_factory and the second should be fixed as the Mermaid docs do not render responsively and should explain how to make a Device class.

Reply:
I agree, but I would rather merge this now and raise separate issues for fixing those pages and consolidating, otherwise this PR will languish for a long time and the published docs (that everyone reads) will continue to contain mistakes that throw people off. Alternatively, I can remove this file entirely from the PR and make issues in dodal, what do you think?

Reply:
https://github.com/DiamondLightSource/dodal/issues/933